src/Repository/ActivityRepository.php

```diff
@@ -30,12 +30,28 @@ public function __construct(ManagerRegistry $registry)
     public function findAuthor($groups)
     {
         return $this->createQueryBuilder('p')
+            ->andWhere('p.isArchived = false')
             ->andWhere('(p.author IN (:groups))')
             ->setParameter('groups', $groups)
             ->orderBy('p.start', 'DESC')
             ->getQuery()
-            ->getResult()
-        ;
+            ->getResult();
+    }
+
+    /**
+     * @param Group[] $groups
+     *
+     * @return Activity[] Returns an array of Activity objects
+     */
+    public function findAuthorArchive($groups)
```

Comment:
This method is identical to the `findAuthor` method except for the isArchived check. My suggestion would be to add a method argument `$archived` instead that's passed as a query parameter, thereby reducing duplicate code.

Reply:
I've thought of this as well. But I feel like passing boolean argument to a function can lead to drastically increasing the technical debt. while it doesn't really do that for a single boolean argument, it does set a president.

The fact that it is such a defined difference that it deserves it's own method. And why I generally agree with keeping the duplication as low as possible, that's part of the reason we have sonarcloud (which, as it seems, doesn't mind me addition)